appinventor/appengine/src/com/google/appinventor/client/TopPanel.java

```diff
@@ -42,29 +49,30 @@
 import java.util.List;
 import java.util.MissingResourceException;
 
-import static com.google.appinventor.client.Ode.MESSAGES;
-
 /**
  * The top panel, which contains the main menu, various links plus ads.
  *
  */
 public class TopPanel extends Composite {
   // Strings for links and dropdown menus:
-  private final DropDownButton accountButton;
-  public DropDownButton languageDropDown;
 
-  private final String WIDGET_NAME_SIGN_OUT = "Signout";
-  private final String WIDGET_NAME_USER = "User";
-  private static final String WIDGET_NAME_LANGUAGE = "Language";
-  private final String WIDGET_NAME_DELETE_ACCOUNT = "DeleteAccount";
+  interface TopPanelUiBinder extends UiBinder<FlowPanel, TopPanel> {}
 
-  private static final String SIGNOUT_URL = "/ode/_logout";
-  private static final String LOGO_IMAGE_URL = "/static/images/codi_long.png";
+  private static final TopPanelUiBinder UI_BINDER = GWT.create(TopPanelUiBinder.class);
 
-  private static final String WINDOW_OPEN_FEATURES = "menubar=yes,location=yes,resizable=yes,scrollbars=yes,status=yes";
-  private static final String WINDOW_OPEN_LOCATION = "_ai2";
+  @UiField(provided = true) FlowPanel header = new FlowPanel("header");
+  @UiField TopToolbar topToolbar;
+  @UiField ImageElement logo;
+  @UiField Label readOnly;
+  @UiField FlowPanel rightPanel;
+  @UiField DropDownButton languageDropDown;
+  @UiField DropDownButton accountButton;
+  @UiField FlowPanel links;
+  private static final String WIDGET_NAME_LANGUAGE = "Language";
+  private final String WIDGET_NAME_DELETE_ACCOUNT = "DeleteAccount";
 
-  private final VerticalPanel rightPanel;  // remember this so we can add MOTD later if needed
+  public static final String WINDOW_OPEN_FEATURES = "menubar=yes,location=yes,resizable=yes,scrollbars=yes,status=yes";
+  public static final String WINDOW_OPEN_LOCATION = "_ai2";
```

Comment:
I'd prefer for static fields to come first in the class definition (with public before private).